packages/neuron-ui/src/components/TransactionFeePanel/index.tsx

```diff
@@ -1,81 +1,143 @@
+    <Button type="text" className={clsx(className, styles['select-item'])} {...res}>
+      <div className={styles.wrap}>
+        <div>
+          <p className={styles.title}>{t(priceObj.text[0])}</p>
+          <p className={clsx(styles[priceObj.color], styles.tag)}>
+            {t('send.price')} {priceObj.text[1]} shannons/kB
+          </p>
+        </div>
+        <p>{sufIcon}</p>
+      </div>
+    </Button>
+  )
+}
+
 const TransactionFee: React.FunctionComponent<TransactionFeeProps> = ({
   price,
   fee,
   onPriceChange,
 }: TransactionFeeProps) => {
   const [t] = useTranslation()
-  const [showDetail, setShowDetail] = useState(false)
+  const [isCustom, setIsCustom] = useState(false)
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false)
+  const dropdownRef = useRef<HTMLDivElement>(null)
+
+  useClickAway(() => {
+    if (isDropdownOpen) {
+      setIsDropdownOpen(false)
+    }
+  }, dropdownRef)
+
+  const selectedPrice = useMemo(() => priceOptions.filter(item => item.key === price)?.[0], [price])
+
+  const onTroggle = (e: any) => {
+    if (isCustom && !selectedPrice) {
+      e.target.value = priceOptions[0].key
+      onPriceChange(e)
+    }
+    setIsCustom(() => !isCustom)
+  }
 
   return (
     <Stack tokens={{ childrenGap: 15 }} aria-label="transaction fee">
       <Stack tokens={{ childrenGap: 15 }}>
-        <TextField label={t('send.fee')} field="fee" value={`${fee} CKB`} readOnly disabled />
-
-        <Toggle
-          onChange={() => {
-            setShowDetail(!showDetail)
-          }}
-          label={t('send.advanced-fee-settings')}
-          inlineLabel
-          onText=" "
-          offText=" "
-          styles={{
-            label: {
-              fontSize: 14,
-              fontWeight: 500,
-            },
-          }}
-        />
+        <TextField label={t('send.fee')} field="fee" value={fee} readOnly disabled width="100%" />
       </Stack>
 
-      <Stack
-        tokens={{ childrenGap: 15 }}
-        styles={{
-          root: {
-            maxHeight: showDetail ? '100vw' : '0',
-            overflow: 'hidden',
-          },
-        }}
-      >
+      {isCustom ? (
         <TextField
-          required
-          label={t('send.price')}
           field="price"
           value={localNumberFormatter(price)}
           onChange={onPriceChange}
           suffix="shannons/kB"
+          width="100%"
+          label={
+            <div className={styles.header}>
+              <div>{t('send.custom-price')}</div>
+              <Button onClick={onTroggle} className={styles['change-btn']} type="text" />
+            </div>
+          }
         />
+      ) : (
+        <div>
+          <div className={styles.header}>
+            <div className={styles.left}>
+              <div>{t('send.price')}</div>
+              <Button type="text" onClick={onTroggle} className={styles['change-btn']} />
+            </div>
+            {/* <div className={styles.right}>单价将在8秒后刷新</div> */}
```

Comment:
```suggestion
```

Reply:
The function of auto-refreshed price will be delivered by https://github.com/nervosnetwork/neuron/pull/2599